runtime/gc_glue_java/MetronomeDelegate.hpp

```diff
@@ -197,305 +197,6 @@ class MM_MetronomeDelegate : public MM_BaseNonVirtual
-			}
-			descriptionBits >>= 1;
-#if defined(J9VM_GC_LEAF_BITS)
-			leafBits >>= 1;
-#endif /* J9VM_GC_LEAF_BITS */
-			if(descriptionIndex-- == 0) {
-				descriptionBits = *descriptionPtr++;
-#if defined(J9VM_GC_LEAF_BITS)
-				leafBits = *leafPtr++;
-#endif /* J9VM_GC_LEAF_BITS */
-				descriptionIndex = J9_OBJECT_DESCRIPTION_SIZE - 1;
-			}
-			scanPtr += 1;
-		}
-
-		env->addScannedBytes(objectSize);
-		env->addScannedPointerFields(pointerFields);
-		env->incScannedObjects();
-
-		return pointerFields;
-	}
-
-	MMINLINE UDATA
-	scanPointerArrayObject(MM_EnvironmentRealtime *env, J9IndexableObject *objectPtr)
-	{
-		UDATA pointerFields = 0;
-		
-#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
-		if(isDynamicClassUnloadingEnabled()) {
-			markClassOfObject(env, (J9Object *)objectPtr);
-		}
-#endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
-
-		bool isContiguous = _extensions->indexableObjectModel.isInlineContiguousArraylet(objectPtr);
-
-		/* Very small arrays cannot be set as scanned (no scanned bit in Mark Map reserved for them) */
-		bool canSetAsScanned = (isContiguous
-				&& (_extensions->minArraySizeToSetAsScanned <= _extensions->indexableObjectModel.arrayletSize(objectPtr, 0)));
-
-		if (canSetAsScanned && _markingScheme->isScanned((J9Object *)objectPtr)) {
-			/* Already scanned by ref array copy optimization */
-			return pointerFields;
-		}
-
-		/* if NUA is enabled, separate path for contiguous arrays */
-		UDATA sizeInElements = _extensions->indexableObjectModel.getSizeInElements(objectPtr);
-		if (isContiguous || (0 == sizeInElements)) {
-			fj9object_t *startScanPtr = (fj9object_t *)_extensions->indexableObjectModel.getDataPointerForContiguous(objectPtr);
-			fj9object_t *endScanPtr = startScanPtr + sizeInElements;
-			pointerFields += scanPointerRange(env, startScanPtr, endScanPtr);
-		} else {
-			fj9object_t *arrayoid = _extensions->indexableObjectModel.getArrayoidPointer(objectPtr);
-			UDATA numArraylets = _extensions->indexableObjectModel.numArraylets(objectPtr);
-			for (UDATA i=0; i<numArraylets; i++) {
-				UDATA arrayletSize = _extensions->indexableObjectModel.arrayletSize(objectPtr, i);
-				/* need to check leaf pointer because this can be a partially allocated arraylet (not all leafs are allocated) */
-				GC_SlotObject slotObject(_javaVM->omrVM, &arrayoid[i]);
-				fj9object_t *startScanPtr = (fj9object_t*) (slotObject.readReferenceFromSlot());
-				if (NULL != startScanPtr) {
-					fj9object_t *endScanPtr = startScanPtr + arrayletSize / sizeof(fj9object_t);
-					if (i == (numArraylets - 1)) {
-						pointerFields += scanPointerRange(env, startScanPtr, endScanPtr);
-						if (canSetAsScanned) {
-							_markingScheme->setScanAtomic((J9Object *)objectPtr);
-						}
-					} else {
-						env->getWorkStack()->push(env, (void *)ARRAYLET_TO_ITEM(startScanPtr));
-					}
-				}
-			}
-		}
-
-		/* check for yield if we've actually scanned a leaf */
-		if (0 != pointerFields) {
-			_scheduler->condYieldFromGC(env);
-		}
```

Comment:
It's hard to follow if this method changed when you moved to RealtimeMarking scheme, but as I minimum, this yield check seems to be lost.

Reply:
condYieldFromGC shouldn't actually be lost since incrementalCompleteScan() regularly does it as objects are scanned? 
 I think this one was just for the leaf slot scan and is now done with a scanner and condYieldFromGC in incrementalCompleteScan()? I'm not sure of a better way